nipoppy/workflows/pipeline.py

```diff
@@ -163,12 +164,86 @@ def fpath_container(self) -> Path:
             )
 
         elif not fpath_container.exists():
-            raise FileNotFoundError(
-                f"No container image file found at {fpath_container} for pipeline"
-                f" {self.pipeline_name} {self.pipeline_version}"
-            )
+            try:
+                self.logger.warning(
+                    f"No container image file found at {fpath_container} for pipeline"
+                    f" {self.pipeline_name} {self.pipeline_version}.\n"
+                )
+                self._get_container_image()
+            except:  # noqa: E722
+                raise FileNotFoundError(
+                    f"No container image file found at {fpath_container} for pipeline"
+                    f" {self.pipeline_name} {self.pipeline_version}"
+                )
         return fpath_container
 
+    def _get_container_image(self):
+        self.logger.info(
+            "Getting container image for "
+            f"{self.pipeline_name} {self.pipeline_version} "
+            f"at {self.fpath_container}"
+        )
+
+        if self.pipeline_name in ["mriqc", "fmriprep"]:
+            docker_hub_repo = "nipreps"
+        elif self.pipeline_name in ["dcm2bids"]:
+            docker_hub_repo = "nipreps"
+        elif self.pipeline_name in ["bidscoin"]:
+            docker_hub_repo = "nipreps"
+        elif self.pipeline_name in ["heudiconv"]:
+            docker_hub_repo = "nipreps"
```

Comment:
Wouldn't it be better to use the existing value in the global config file for this?
```python
self.pipeline_config.CONTAINER_INFO.URI
```

That way, this would work even for pipelines that do not come with Nipoppy by default.

Reply:
Yeah I was planning to refactor things once I got the tests to pass.
But right now I am stuck on the red part of the "red - green - refactor" of test driven development.